### PR TITLE
[Ansible] Support multiple passwords for SSH connection

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Add connection plugin `multi_passwd_ssh` to support specify
an alternative SSH password via setting anyone of the following
three Ansible variables:
```
ansible_altpassword
ansible_ssh_altpass
ansible_ssh_altpassword
```
With this, Ansible will try to authenticate with the password specified
by `ansible_password` first, if the authentication fails, it will switch
to `ansible_altpassword`.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
To test `SONiC` from different vendors with different admin passwords.

#### How did you do it?
Add decorator `_password_retry` to catch `AnsibleAuthenticationFailure` raised from Ansible SSH connection and switch
to alternative password to try.

#### How did you verify/test it?
1. Ansible
```
$ ansible -i str -m ping -e "ansible_connection='multi_passwd_ssh'" -e "ansible_password='wrongpass'" -e "ansible_ssh_altpass='correctpass'" lab-msn2700-2
lab-msn2700-2 | SUCCESS => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "ping": "pong"
}
```
2. Pytest
```
# export ANSIBLE_LIBRARY=../ansible
# export ANSIBLE_CONNECTION_PLUGINS=../ansible/plugins/connection
# pytest --inventory ../ansible/lab --host-pattern all -vvv --show-capture stdout --testbed vms1-t0 --testbed_file ../ansible/testbed.csv --skip_sanity --log-file-level debug -vvvvv bgp/test_bgp_fact.py
```

### CODE REVIEW

File: ansible/plugins/connection/sonic_ssh.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import imp
```
Comment (lguohan) [inline ansible/plugins/connection/sonic_ssh.py:1]: better to rename the file as multi_ssh or multi_passwd_ssh instead of sonic_ssh?

Comment (lolyu) [reply ansible/plugins/connection/sonic_ssh.py:1]: Sure.

**Comment by lguohan** [comment]
i am curious if we can contribute this to ansible and it seems very generic feature for ansible.

**Comment by lolyu** [comment]
> i am curious if we can contribute this to ansible and it seems very generic feature for ansible.

Sure, I will take this into consideration, but as I'm concerned, someone might argue that the correctness of SSH password should be the responsibility of Ansible user instead of making Ansible take a guess. I will raise an issue regarding this to see the opinion from the Ansible community.

**Pull request merged**